src/Microsoft.ML.Experimental/DataLoadSave/Database/DatabaseLoader.cs

```diff
@@ -98,6 +102,70 @@ void ICanSaveModel.Save(ModelSaveContext ctx)
         /// <param name="source">The source from which to load data.</param>
         public IDataView Load(DatabaseSource source) => new BoundLoader(this, source);
 
+        internal static DatabaseLoader CreateDatabaseLoader<TInput>(IHostEnvironment host)
+        {
+            var userType = typeof(TInput);
+
+            var fieldInfos = userType.GetFields(BindingFlags.Public | BindingFlags.Instance);
+
+            var propertyInfos =
+                userType
+                .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                .Where(x => x.CanRead && x.GetGetMethod() != null && x.GetIndexParameters().Length == 0);
+
+            var memberInfos = (fieldInfos as IEnumerable<MemberInfo>).Concat(propertyInfos).ToArray();
+
+            if (memberInfos.Length == 0)
+                throw host.ExceptParam(nameof(TInput), $"Should define at least one public, readable field or property in {nameof(TInput)}.");
```

Comment:
I wonder if this is a place we can deviate from TextLoader. The user shouldn’t be required to give a column index/order here. Instead, we can look the column index up given the column name. We already have support for that in the DatabaseLoader. We should simply be able to remove this exception and let that behavior kick in.

Reply:
See 

https://github.com/dotnet/machinelearning/blob/e389752920ef0f58c5d3f6890d3609f7f8510df1/src/Microsoft.ML.Experimental/DataLoadSave/Database/DatabaseLoaderCursor.cs#L314-317